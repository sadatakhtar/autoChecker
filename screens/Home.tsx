import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { UseSelector, useSelector } from 'react-redux'
import { getTest } from '../src/features/general/generalSlice'


type Props = {}

const Home = (props: Props) => {
    const testRedux = useSelector(getTest);
  return (
    <View>
      <Text>Home</Text>
      <View>
        <Text style={{fontSize: 20, color: 'red'}}>{testRedux}</Text>
      </View>
    </View>
  )
}

export default Home

const styles = StyleSheet.create({})