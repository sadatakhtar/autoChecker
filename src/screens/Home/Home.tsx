import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import StandardHeader from '../../components/StandardHeader';
import Btn from '../../components/Btn';
import {useNavigation} from '@react-navigation/native';
import Paragraph from '../../components/Paragraph';

interface HomeProps {}
interface NavProps {
  navigate: (arg0: string) => void;
}

const Home = ({}: HomeProps) => {
  const navigation: NavProps = useNavigation();

  const handleBtn = () => {
    console.log('btn pressed');
    navigation.navigate('Dashboard');
  };
  return (
    <View style={styles.container}>
      <StandardHeader />
      <View style={styles.body}>
        <Paragraph
          summary="Welcome to AutoChecker, your ultimate companion for stress-free car
        buying! Are you a first-time car buyer feeling overwhelmed by the
        uncertainties of purchasing a vehicle? Or perhaps you're a cautious
        shopper who wants to ensure you're making a wise investment? Look no
        further!"
        />
        <Paragraph
          summary="AutoChecker is designed with you in mind. 
        Our goal is to empower you with 
        knowledge, helping you make informed decisions and avoid unexpected 
        surprises down the road."
        />
        <Paragraph
          summary="Say goodbye to buyer's remorse and hello to 
        confidence with AutoChecker. Start your journey towards a 
        worry-free car buying experience today!"
        />
        <View>
          <Btn
            title="Click to proceed"
            onPress={handleBtn}
            style={{marginTop: 20}}
          />
        </View>
      </View>
    </View>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    //padding: 10,
  },
  text: {
    fontSize: 18,
  },
  body: {
    padding: 10,
    marginVertical: 20,
  },
});
