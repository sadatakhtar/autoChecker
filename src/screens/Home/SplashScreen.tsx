import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import {colours} from '../../../assets/SharedStyles';
import * as Animatable from 'react-native-animatable';

interface SplashScreenProps {
  //navigation: undefined | any;
}

const SplashScreen = ({}: SplashScreenProps) => {
 
  return (
    <View style={styles.container}>
      <View style={styles.innerWrapper}>
        <Animatable.Text
          style={styles.text}
          duration={2000}
          animation="fadeInDownBig">
          Auto<Text style={{color: colours.$yellow}}>C</Text>hecker
        </Animatable.Text>

        <Animatable.Text
          style={styles.subText}
          duration={3000}
          animation="lightSpeedIn">
          Your ultimate car companion
        </Animatable.Text>
      </View>
    </View>
  );
};

export default SplashScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colours.$black,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: colours.$white,
    fontSize: 40,
    fontWeight: 'bold',
  },
  subText: {
    color: colours.$white,
    fontSize: 20,
    fontWeight: 'bold',
  },
  innerWrapper: {
    justifyContent: 'center',
    alignItems  : 'center',
  }
});
