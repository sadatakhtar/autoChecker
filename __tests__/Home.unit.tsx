import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Home from '../screens/Home'; 

describe('Home Component Tests', () => {
  // Test 1: Component renders without crashing
  it('renders without crashing', () => {
    render(<Home />);
    expect(screen).not.toBeNull(); 
  });

  // Test 2: Check for welcome text
  it('displays the correct welcome text', () => {
    render(<Home />);
    const welcomeText = screen.getByText(/Welcome to AutoChecker, enter a vehicle registration to get a list of possible defects or issues which can surface in the near future./i);
    expect(welcomeText).toBeTruthy();
  });
});