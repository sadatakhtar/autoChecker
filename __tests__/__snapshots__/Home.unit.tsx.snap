// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen Snapshot Test renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#99ccff",
        "borderColor": "#1f446d",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Text>
        LOGO
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Auto
        <Text
          style={
            {
              "color": "#4c93ff",
            }
          }
        >
          C
        </Text>
        hecker
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 20,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "marginVertical": 6,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
          }
        }
      >
        Welcome to AutoChecker, your ultimate companion for stress-free car buying! Are you a first-time car buyer feeling overwhelmed by the uncertainties of purchasing a vehicle? Or perhaps you're a cautious shopper who wants to ensure you're making a wise investment? Look no further!
      </Text>
    </View>
    <View
      style={
        {
          "marginVertical": 6,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
          }
        }
      >
        AutoChecker is designed with you in mind.  Our goal is to empower you with  knowledge, helping you make informed decisions and avoid unexpected  surprises down the road.
      </Text>
    </View>
    <View
      style={
        {
          "marginVertical": 6,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
          }
        }
      >
        Say goodbye to buyer's remorse and hello to  confidence with AutoChecker. Start your journey towards a  worry-free car buying experience today!
      </Text>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#99ccff",
                "borderColor": "#1f1f1f",
                "borderRadius": 5,
                "borderWidth": 2,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
              },
              {
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#1f1f1f",
                  "fontSize": 20,
                  "fontWeight": "condensedBold",
                },
                undefined,
              ]
            }
          >
            Click to proceed
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
